src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -132859,6 +132877,45 @@ public static void Main()
             CompileAndVerify(executeComp, expectedOutput: "ran");
         }
 
+        [Fact, WorkItem(41437, "https://github.com/dotnet/roslyn/issues/41437")]
+        public void CannotAssignMaybeNullToTNotNull()
+        {
+            var source = @"
+class C
+{
+    TNotNull M<TNotNull>(TNotNull t) where TNotNull : notnull
```

Comment:
Sorry, I should have asked earlier, but do corresponding tests already exist for 'class' constraint? (I would be surprised if they didn't..)

Reply:
Yes, it's covered by `NullCastToUnannotatableReferenceTypedTypeParameter` I think